dmci/api/app.py

```diff
@@ -230,7 +231,8 @@ def _handle_persist_file(status, full_path, reject_path=None, reject_reason=""):
 
         else:
             try:
-                os.rename(full_path, reject_path)
+                shutil.copy(full_path, reject_path)
+                os.remove(full_path)
             except Exception as e:
                 logger.error("Failed to move persist file to rejected folder: %s", reject_path)
                 logger.error(str(e))
```

Comment:
The unit test fails because now we have to parts of code inside the same function which can possibley throw OSError exceptions

first it is shutil.copy() in the first try-clause, and then it is "with open" to write reason file.

So the unittests fails, and I cannot see how I can test OSError exception both for the shutil.copy() AND the open()